libs/beacon/src/beacon_service.cpp

```diff
@@ -349,6 +376,28 @@ BeaconService::State BeaconService::OnCompleteState()
                                     block_entropy_previous_->EntropyAsSHA256(),
                                     block_entropy_being_created_->group_signature));
 
+  // Trim maps of unnecessary info
+  {
```

Comment:
Might be cleaner to use your fixed size structure here although it would need to do an iteration for the lookup